### PR TITLE
30 result box integration

### PR DESCRIPTION
**Pull request by xgarciagSopra**
CSS casillas y corrección de css de selección de casilla

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,33 +1,42 @@
-import { TestBed } from '@angular/core/testing';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { RouterTestingModule } from '@angular/router/testing';
 import { AppComponent } from './app.component';
+import { HeaderComponent } from './components/header/header.component';
+import { GameCenterComponent } from './components/game-center/game-center.component';
+import { FooterComponent } from './components/footer/footer.component';
+import { MatDialog } from '@angular/material/dialog';
+
+xdescribe('AppComponent', () => {
```
Comment (ddiezgsop) [inline src/app/app.component.spec.ts]:
```suggestion
describe('AppComponent', () => {
```

xdescribe?

File: src/app/components/game-center/game-center.component.spec.ts [hunk 2]
```diff
@@ -20,4 +38,8 @@ describe('GameCenterComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  // it('should check', () => {
+  //   expect(component).toBeTruthy();
+  // });
```
Comment (ddiezgsop) [inline src/app/components/game-center/game-center.component.spec.ts]:
```suggestion
   it('should check', () => {
     expect(component).toBeTruthy();
   });
```

Do not upload commented code

File: src/app/components/result-box/result-box.component.scss [hunk 3]
```diff
@@ -13,3 +13,24 @@
...
+    color: white;
+  }
+}
+
+.fail {
+  background-color: $fail;
+  p {
+    color: white;
+  }
+}
```
Comment (ddiezgsop) [inline src/app/components/result-box/result-box.component.scss]: Use a class in `<p>` to give it `color: white` and change the background-color when u need. Notice that u are repeating code 3 times

File: src/app/components/result-box/result-box.component.ts [hunk 4]
```diff
@@ -9,16 +9,22 @@ export class ResultBoxComponent {
   @Input() word = '';
   @Input() splitWord: string[] = [];
   @Input() borderResultStyles!: boolean;
+  @Input() keyBoxStyle: string = '';
```
Comment (ddiezgsop) [inline src/app/components/result-box/result-box.component.ts]:
```suggestion
  @Input() keyBoxStyle = '';
```

File: src/app/components/result-box/result-box.component.ts [hunk 5]
```diff
@@ -9,16 +9,22 @@ export class ResultBoxComponent {
...
     this.selectResultBox.emit(boxIndex);
   }
+
+  newBoxStyles(letter: string, index: number) {
+    return {
+      '': !this.firstRound,
+      hit: this.keyBoxStyle === 'hit',
+      parcialHit: this.keyBoxStyle === 'parcialHit',
+      fail: this.keyBoxStyle === 'fail',
+      'border-primary': this.selectedBox === index,
```
Comment (ddiezgsop) [inline src/app/components/result-box/result-box.component.ts:26]:
```suggestion
      'selectedSquare': this.selectedBox === index,
```

Use a more clarifing variable

File: src/app/components/result-box/result-box.component.ts [hunk 6]
```diff
@@ -9,16 +9,22 @@ export class ResultBoxComponent {
...
-    return this.borderResultStyles ? 'border-success' : 'border-danger';
-  }
-
   selectedResultBox(boxIndex: number) {
     this.selectResultBox.emit(boxIndex);
   }
+
+  newBoxStyles(letter: string, index: number) {
+    return {
+      '': !this.firstRound,
```
Comment (ddiezgsop) [inline src/app/components/result-box/result-box.component.ts]:
```suggestion
```

Don't use this case, if u don't give any styles, don't give any condition

File: src/app/services/guess-word.service.spec.ts [hunk 7]
```diff
@@ -1,12 +1,15 @@
...
 import { GuessWordService } from './guess-word.service';
+import { HttpClient, HttpHandler } from '@angular/common/http';
 
 describe('GuessWordService', () => {
   let service: GuessWordService;
 
   beforeEach(() => {
-    TestBed.configureTestingModule({});
+    TestBed.configureTestingModule({
+      providers: [HttpClient, HttpHandler],
```
Comment (ddiezgsop) [inline src/app/services/guess-word.service.spec.ts]:
```suggestion
      imports: [HttpClientTestingModule],
```

When u are testing the http service, u need to import the HttpTestingModule in imports

**Review by ddiezgsop — CHANGES_REQUESTED** [review]

File: src/app/app.component.spec.ts (hunk 1 above)
Comment (xgarciagSopra) [reply src/app/app.component.spec.ts]: Lo pusiste tu, cuando me tocaste el spec, no he tocado nada

File: src/app/components/result-box/result-box.component.ts (hunk 5 above)
Comment (xgarciagSopra) [reply src/app/components/result-box/result-box.component.ts:26]: es de boostrap, si pongo el selectedSquare no me deja poner un borde con color de normal dentro de css

File: src/app/app.component.spec.ts (hunk 1 above)
Comment (ddiezgsop) [reply src/app/app.component.spec.ts]: Remove it, xdescribe and fdescribe shouldn't be uploaded

File: src/app/components/result-box/result-box.component.scss [hunk 8]
```diff
@@ -13,3 +13,19 @@
...
   }
 }
+
+.hit {
+  background-color: $hit;
+}
+
+.colorWhite {
+  color: white;
+}
```
Comment (ddiezgsop) [inline src/app/components/result-box/result-box.component.scss]:
```suggestion
.baseSquare {
  color: white;
}
```

**Review by ddiezgsop — APPROVED** [review]

**Pull request merged**